spec-tests/src/test/java/org/candlepin/spec/CloudRegistrationSpecTest.java

```diff
@@ -587,8 +587,54 @@ public void shouldCreateConsumerWhenUsingAnonymousToken() throws JsonProcessingE
     }
 
     @Test
-    public void shouldCreateOwnerWhenNoOwnerExistsForCloudAccountDuringV2Auth()
-        throws JsonProcessingException {
+    @OnlyInHosted
+    public void shouldMigrateConsumersRegisteredAfterClaiming() {
+        ApiClient admin = ApiClients.admin();
+        OwnerClient owners = admin.owners();
+        OwnerDTO anonOwner = Owners.randomSca().anonymous(true);
+        OwnerDTO destOwner = owners.createOwner(Owners.randomSca());
+
+        ProductDTO productDTO = Products.random();
+
+        String accountId = StringUtil.random("cloud-account-id-");
+        String instanceId = StringUtil.random("cloud-instance-id-");
+        String offerId = StringUtil.random("cloud-offer-");
+
+        admin.hosted().createProduct(productDTO);
+        admin.hosted().associateProductIdsToCloudOffer(offerId, List.of(productDTO.getId()));
+
+        admin.hosted().createOwner(anonOwner);
+        admin.hosted().associateOwnerToCloudAccount(accountId, anonOwner.getKey());
+        admin.hosted().createSubscription(Subscriptions.random(anonOwner, productDTO));
+        admin.owners().createOwner(anonOwner);
+        admin.ownerProducts().createProduct(anonOwner.getKey(), productDTO);
+        admin.owners().createPool(anonOwner.getKey(), Pools.random(productDTO));
+
+        CloudAuthenticationResultDTO result = ApiClients.noAuth().cloudAuthorization()
+            .cloudAuthorizeV2(accountId, instanceId, offerId, "test-type", "");
+        ApiClient anonClient = ApiClients.bearerToken(result.getToken());
```

Comment:
Because the org was fully ready before you called the cloudAuthorizeV2, the token you're getting here isn't anonymous, its the standard one. Its good we are testing this, but we also want to test that clients that had an anonymous token can also register after the claiming is done.
You can add another `cloudAuthorizeV2` call right before the `admin.owners().createPool()` call, which should return an anonymous token, and test registering with that too after the claiming

Reply:
Done.

Reply:
Now it is creating an anonClient, but it still needs to use it to register a consumer after the claiming job is finished

Reply:
Done.